Recruit-Training/Advanced-Recruit-Training/src/LogFileReader.cpp

```diff
@@ -57,15 +58,30 @@ bool LogFileReader::readAll(const QString& fileName)
  * that the conversion from string to double is sucessful.*/
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
+
     QStringList sections = line.split(BATDATA_DELIMITER);
 
+    if(sections.length() != COLUMNS){
+        return false;
+    }
+
     QString timeString = sections.at(0);
-    batteryData.time = QTime::fromString(timeString, STRING_TIME_FORMAT);
 
-    batteryData.voltage = sections.at(1).toDouble();
 
-    batteryData.current = sections.at(2).toDouble();
 
-    return true;
+    QTime time1 = QTime::fromString(timeString, STRING_TIME_FORMAT);
+    batteryData.time = time1;
+    bool isTimeValid = QTime::isValid(time1.hour(), time1.minute(), (time1.second() + (time1.msec()/1000)));
+    bool flag1;
+    bool flag2;
```

Comment:
Initialize the variables to show there explicit beginning state. Also use descriptive names